src/components/Modal/index.tsx

```diff
@@ -35,9 +42,20 @@ function Modal({fullscreen = true, onModalHide = () => {}, type, onModalShow = (
             setStatusBarColor(isFullScreenModal ? theme.appBG : StyleUtils.getThemeBackgroundColor(statusBarColor));
         }
 
+        if (shouldHandleNavigationBack) {
+            window.history.pushState({shouldGoBack: true}, '', null);
+            window.addEventListener('popstate', handlePopStateRef.current);
+        }
```

Comment:
This addition contributed to creating the following issue:
- https://github.com/Expensify/App/issues/44986

**Explanation:** (from [this](https://github.com/Expensify/App/issues/44986#issuecomment-2216891381) proposal RCA)
This `popstate` listener which will close the modal when triggered if `shouldHandleNavigationBack` is true (it's only true for emoji picker).

https://github.com/Expensify/App/blob/dd96852320612fd5188c95e659fce3814150af82/src/components/Modal/index.tsx#L46-L49

It's triggered when we call `back` here when the modal is closed if the state contains `shouldGoBack` property that we push above.

https://github.com/Expensify/App/blob/dd96852320612fd5188c95e659fce3814150af82/src/components/Modal/index.tsx#L23-L29

However, the `back` is triggered even though we don't close the emoji picker. That's because it's triggered from the context menu hide. When we show the emoji picker, the context menu is closed which triggers the `back`. `shouldGoBack` exists in the history state because the emoji picker is shown first which already pushes the new state that sets `shouldGoBack` to true.

Long-press comment -> context menu shown -> press add emoji reaction -> push state, shouldGoBack: true -> context menu hides -> history.back -> triggers popstate -> emoji picker closes.